app/models/identification.rb

```diff
@@ -179,8 +180,8 @@ def set_previous_observation_taxon
     end
     unless previous_observation_taxon_id
       working_created_at = created_at || Time.now
-      previous_ident = observation.identifications.select{|i| i.persisted? && i.current && i.created_at < working_created_at }.last
-      previous_ident ||= observation.identifications.select{|i| i.persisted? && i.created_at < working_created_at }.last
+      previous_ident = observation.identifications.by(user_id).older_than(working_created_at).current.select(&:persisted?).last
+      previous_ident ||= observation.identifications.by(user_id).older_than(working_created_at).select(&:persisted?).last
```

Comment:
Again, not blocking, but using scopes here means that if the second line runs here, you will query the database a second time. `observation.identifications.select` will hit the database once (or maybe not at all if it was already called) and load all the observation's identifications into memory, and all subsequent calls to `.select` will probably be faster than their database-based equivalents. All of the Identification callbacks could probably use an audit to minimize db queries, and I realize I'm picking nits here, but identification creation is a high-volume, low performance operation as it is.

Reply:
Hmm, so this is not something I've benchmarked (or probably something I realistically could in a dev environment) but my initial thought was kind of the opposite. 

I was thinking that by delegating the filtering to SQL we could avoid hydrating many instances that would be immediately thrown out by the select. That said, yeah, since the `persisted?` call will resolve the AR relation anyway it is possible to require two queries. Also, now that I'm thinking about it more, the number of associated identifications per observation is probably not great enough to really see an issue with loading them all into memory/deserializing/etc. 

I'll change it back.